models/Asset/Thumbnail/ImageThumbnailTrait.php

```diff
@@ -310,22 +310,54 @@ public function getMimeType()
      */
     public function getFileExtension()
     {
-        return \Pimcore\File::getFileExtension($this->getPath(true));
+        return \Pimcore\File::getFileExtension($this->getPath());
+    }
+
+    /**
+     * TODO: Pimcore 11: remove convertArgsBcLayer method (BC layer)
+     *
+     * @param array $args
+     *
+     * @return array
+     */
+    protected function convertArgsBcLayer($args)
+    {
+        if ((count($args) == 1 && is_array($args[0])) || count($args) == 0) {
+            return $args[0] ?? [];
+        } else {
+            trigger_deprecation('pimcore/pimcore', '10.6',
+                'Calling the getPath() method with arguments is deprecated since version 10.6 and will be removed in Pimcore 11.
+                Use an array with options (e.g. ->getPath(["deferredAllowed" => true, "cacheBuster" => false]))');
+
+            if (count($args) == 1) {
+                return [
+                    'deferredAllowed' => $args[0],
+                ];
+            } elseif (count($args) == 2) {
+                return [
+                    'deferredAllowed' => $args[0],
+                    'cacheBuster' => $args[1],
+                ];
+            }
+        }
+
+        return [];
```

Comment:
It's fine but would like to suggest some small refactor.
The first IF block looks a bit "overcrowded" with the coalescing, and the `return []` at the bottom is only valid when $args are more than 2 and at the same time it has to stay to avoid (i guess) phpstan error.

```suggestion
        $totalArgs = count($args);
        if ($totalArgs > 0) {
            if (is_array($args[0])){
                return $args[0];
            }

            trigger_deprecation('pimcore/pimcore', '10.6',
                'Calling the getPath() method with arguments is deprecated since version 10.6 and will be removed in Pimcore 11.
            Use an array with options (e.g. ->getPath(["deferredAllowed" => true, "cacheBuster" => false]))');

            if ($totalArgs === 1){
                return [
                    'deferredAllowed' => $args[0],
                ];
            }
            return [
                'deferredAllowed' => $args[0],
                'cacheBuster' => $args[1],
            ];
        }

        return [];
```

Reply:
@kingjia90 thank you. you're right. looks cleaner!